_posts/2021-08-23-gsoc-2021-project.md

```diff
@@ -0,0 +1,208 @@
+theme: twitter
+---
+
+{% include JB/setup %}
+
+## Executive Summary
+
+Here's an update on new features related to visualizing Dask graphs and HTML representations. You can try these new features today with version `2021.08.1` or above. This work was done by Freyam Mehta during the Google Summer of Code 2021. Dask took part in the program under the NumFOCUS umbrella organization.
+
+## Contents
+
+- [Visualizing Dask graphs](#visualizing-dask-graphs)
+    - [Graphviz node size scaling](#graphviz-node-size-scaling)
+    - [New tooltips](#new-tooltips)
+    - [Color by layer type](#color-by-layer-type)
+    - [Bugfix in visualize method](#bugfix-in-visualize-method)
+- [HTML Representations](#html-representations)
+    - [Array images in HTML repr for high level graphs](#array-images-in-html-repr-for-high-level-graphs)
+    - [New HTML repr for ProcessInterface class](#new-html-repr-for-processinterface-class)
+    - [New HTML repr for Security class](#new-html-repr-for-security-class)
+
+## Visualizing Dask graphs
+
+There are several new features involving Dask graph visualization. Freyam worked on making them more illustrative, engaging, and informative. The Graphviz library boasts a great set of attributes which can be modifified to create a more visually appealing output. 
+
+These features primarily improve the Dask high level graph visualizations. Both low level and high level Dask graphs can be accessed with very similar methods:
+
+- Dask low level graph: `result.visualize()`
+- Dask high level graph: `result.dask.visualize()`
+
+### Graphviz node size scaling
+
+The first change you may notice to the Dask high level graphs, is that the node sizes have been adjusted to scale with the number of tasks in each layer. Layers with more tasks would appear larger than the rest.
+
+This is a helpful feature to have, because now users can get a much more intuitive sense of where the bulk of their computation takes place.
+
+Example:
+
+```python
+import dask.array as da
+
+array = da.random.random((10000, 10000), chunks=(200, 200))
+result = array + array.T - array.mean(axis=0)
+
+result.dask.visualize()  # Dask high level graph
+```
+
+<img src="/images/gsoc21/7869.png" alt="Example: graphviz node size scaling, pull request #7869" height=414 width=736>
+
+Note: this change only affects the graphviz output for Dask high level graphs. Low level graphs are left unchanged.
+
+Reference: [Pull request #7869 by Freyam Mehta *"Add node size scaling to the Graphviz output for the high level graphs"*](https://github.com/dask/dask/pull/7869)
+
+### New tooltips
+
+Dask high level graphs now include hover tooltips to provide a brief summary of more detailed information. To use the tooltips, generate a dask high level graph (eg: `result.dask.visualize()`) then hover your mouse above the layer you are interested in.
+
+<img src="/images/gsoc21/7973.png" alt="Example: tooltips provide extra information, pull request #7973" height=414 width=736>
```

Comment:
Nice to have, but not required: it would be good if this demo image used the same example code as the node scaling demo in the section above. Then this section feels more like a natural continuation.

@freyam